src/database/subClasses/ExperimentMethods.java

```diff
@@ -260,6 +263,11 @@ public int updateExperiment(String expID, HashMap<String, String> annotations)
 
             int arr[] = stmt.executeBatch();
             for (int i = 0; i < arr.length; i++) {
+                if (arr[i] == 0) {
+                    String label = keys.get(i);
+                    String value = annotations.get(label);
+                    arr[i] = annotateExperiment(expID, label, value);
```

Comment:
`annotateExperiment` may throw `IOException`s. You may want to catch those as well.